src/services/namingService.ts

```diff
@@ -0,0 +1,160 @@
+import { ServerlessAzureConfig, ResourceConfig } from "../models/serverless"
+import { Guard } from "../shared/guard"
+import md5 from "md5";
+import configConstants from "../config";
+
+export class AzureNamingService {
+
+  /**
+   * Get a resource name for an Azure Service. Naming convention:
+   *
+   * {prefix}-{shortRegionName}-{shortStageName}(optionally: -{suffix})
+   *
+   * @param config Serverless Azure Config for service (serverless.service)
+   * @param resourceConfig
+   * @param suffix
+   */
+  public static getResourceName(config: ServerlessAzureConfig, resourceConfig?: ResourceConfig, suffix?: string) {
+    if (resourceConfig && resourceConfig.name) {
+      return resourceConfig.name;
+    }
+    const { prefix, region, stage } = config.provider
+    let name = [
+      prefix,
+      this.createShortAzureRegionName(region),
+      this.createShortStageName(stage),
+    ].join("-");
+    if (suffix) {
+      name += `-${suffix}`;
+    }
+    return name;
```

Comment:
I can't think of a _great_ reason why, but would we want to `.toLowerCase()` this, just to normalize?  Also, guessing it's unsupported, but has anyone tried extended ascii or double-byte chars?